app/controllers/courses_controller.rb

```diff
@@ -3,6 +3,7 @@
 class CoursesController < ApplicationController
   before_action :set_course, only: %i[show edit update destroy]
   before_action :set_subject
+  before_action :set_subjects
```

Comment:
Are we doing this for every Controller? If so, maybe we should move it to the ApplicationController, along with the method itself.